scripts/xe-enable-ipv6

```diff
@@ -32,8 +32,8 @@ if [ $1 = "enable" ]; then
 	echo "IPV6_AUTOCONF=NO" >> /etc/sysconfig/network
 	chkconfig ip6tables on
 
-	sysctl -q -w "net.ipv6.conf.all.disable_ipv6=0"
-	sysctl -q -w "net.ipv6.conf.default.disable_ipv6=0"
+	echo "net.ipv6.conf.all.disable_ipv6=0" >> /etc/sysctl.d/91-net-ipv6.conf
```

Comment:
Shouldn't this be just `>` to write on top of the existing file instead of appending the contents? (same in line 56)

Reply:
I used sed to remove the entries.
So that a `net.ipv6.conf.<iface>.disable_ipv6` entry can be put and not erased.

Reply:
I think I should do the same sed command for `90-net.conf`  to be sure that the `disable_ipv6` entries are removed from this file?

Reply:
It's easier to reason about this if every time the contents of the file are written instead of searching code to delete then append contents.

Unless I've missed something this code is the only code that writes to this file, there's no point in being "nice" and expect that the file might have more contents.

Reply:
I guess you're right yeah.
I'll change that then.
I think I still need to seek and erase the entry in `90-net.conf`

Reply:
I don't see any mentions in 90-net.conf here. Is it coming from the default centos configuration?

Reply:
On CH:
```
# rpm -qf /etc/sysctl.d/90-net.conf 
xenserver-release-config-8.2.50-8.xs8.x86_64
# tail -n3 /etc/sysctl.d/90-net.conf 
# Disable IPv6 on interfaces
net.ipv6.conf.all.disable_ipv6 = 1
net.ipv6.conf.default.disable_ipv6 = 1
```

Reply:
By default, on my host, IPv6 is disabled in /etc/sysctl.d/90-net.conf, while the kernel ipv6 module is still loaded. So I don't think that we need that  /etc/modprobe.d/disable-ipv6 stuff anymore?

And this PR now seems to create a 91-net.conf file that conflicts with 90-net.conf?

Reply:
The modprobe thing is probably useless yes.
This PR creates a new config file IPv6 specific yes.
Indeed for now it conflicts with 90-net.conf, I'm currently searching where the 90-net.conf ipv6 entries are written in the file.

Reply:
For Citrix Hypervisor, the 90-net.conf file is just a static file in the xenserver-release-config RPM.

Reply:
And it's the same in XCP-ng, only the name of the RPM changes.

It's defined as `%config(noreplace)` so once put there it can be altered by users or anything without being replaced in future updates, except ISO upgrades.

Should the installer and/or XAPI modify it when needed to enable ipv6?

Or should the defaults be different at installation, and/or the ipv6-related settings extracted to a separate configuration file so that we can let the installer/XAPI rewrite it entirely when needed?

Reply:
Perhaps splitting up those conf files is the best here. So in xenserver-release-config and the xcp-ng equivalent, move the IPv6 lines from 90-net.conf to 91-net-ipv6.conf, and then this current PR seems to do the right thing (but just delete the modprobe stuff while we are here).

Reply:
This looks like a good solution to me.

Reply:
LGTM as well, i'll do the change for modprobe and we'll change the config file in our rpm :)

Thanks for the help!